### DISCUSSION TITLE
How to redefine size_t in coc-clang?

### DISCUSSION THREAD

**Discussion by joseph-pq** [neoclide/coc.nvim#3811] [Q&A]
Hey guys!
My environment is the following:
- ubuntu 20.04
- nvim v0.7.0
- coc with coc-clangd
- clang 15.0.0
I'm working with esp32 in esp-idf , they have a complete environment where they use FreeRTOS.
When I compile my project using cmake,  I enable the compile_commands flag in order to have the help of clang while coding, that is, the warnings and errors coc-clangd puts in nvim while I'm editing.
The problem I have is the following:
The size_t in esp-idf is an unsigned of 32bits. For some reason, coc-clang doesn't know that and rises me an error when i try to print a size_t variable using %u in the printf instead of %lu. 
The mesage is:
Format specifies type 'unsgined int' but the argument has type 'size_t' (aka 'unsigned long')
[clang: -Wformat]
Does anyone know how I can fix that?

**Comment by chemzqm** [comment]
Configure your clangd might helps
